charts/consul/templates/client-snapshot-agent-deployment.yaml

```diff
@@ -137,10 +147,17 @@ spec:
           {{- .Values.client.snapshotAgent.caCert | nindent 14 }}
           EOF
           {{- end }}
+          {{- if  .Values.global.secretsBackend.vault.enabled }}
+          cat /vault/secrets/snapshot-agent-config.txt | base64 -d > /vault/secrets/snapshot-agent-config.json
```

Comment:
consul snapshot command will look for only files with a .json extension and the contents from vault need to be base64 decoded.